course-content/13_working_with_code.qmd

```diff
@@ -0,0 +1,397 @@
+    <p style="text-align: center">
+    
+</left>
+
+In the example below, the issue is not a syntax error (although it might like a typo that one might associate with being a syntax error)
+<left>
+    <img src="images/working-with-code/runtime_error2.png" size=500 style="margin:auto" border=1/>
+    <p style="text-align: center">     
+  
+</left>
+
+The above examples introduce the concept of *traceback*, which is a way of helping the coder locate the origin of the error experienced. A *traceback* will show up in output that is to be read "from the bottom up".  In the above example, we see that the program does not recognize the object "aege" when it encounters it in cell line 2. Note that this does not necessarily mean that the problem should be fixed at the location indicated. In the above example, a reexamination of the code might reveal that the original name assigned was incorrectly spelled. Were that the case, then the original name assignment should be updated, not the reference to it in cell line 2. 
+
+
+*Diving deeper*
+
+The first and best way to work though why your code failed is to look at the information provided at the bottom of the output. As in the examples above, much of the time you will have enough information there to give you some ideas about what the cause of the problem is and a starting point for modifying the code and running it again. 
+
+Many times, however, you will find that your code throws errors you still don't understand.  Other times, you would like to get a better understanding of some topic you have come across. For those moments, don't be shy to use Google to get going by typing in the search field those terms that are narrow enough to describe your problem as succinctly as possible.
+
+<center>
+    <img src="images/working-with-code/goog-stackoverflow.png" size=700 style="margin:auto" border=1/>
+    <p style="text-align: center">
+        Global community and solutions to challenges or fixes for bugs
+    </p>
+</center>
+
+Among the more popular entries you'll find on the search engine response page come from stackoverflow or other coding community sites, and from blogs or other sources. It may also pay to look at "videos" among your Google search responses and you will likely see many YouTube offerings of varying degrees of relevance. There is a whole ecosystem of python developers (polished and not-so-much) who offer content that may be of help. These are often very helpful. 
+
+Imagine we were working through the "SyntaxError: EOL while scanning string literal" error above. You know that it's a syntax error, so you examine your code for something that does not look right based on what you know about writing code. What happens if you still are getting an error? Then a Google search might be helpful.
+
+
+<center>
+    <img src="images/working-with-code/google_syntax_error.png" size=500 style="margin:auto" border=1/>
```

Comment:
If you use width="100%" (or some other %) instead of size=500 it will use relative sizing so that the image is guaranteed to fall within the confines of the column width. (On my screen this image hangs off to the right and overlaps with the table of contents)

![image](https://user-images.githubusercontent.com/45674978/202321297-a569732f-5a19-41db-999b-f904e32e255a.png)

Reply:
Thanks! It is weird to me that it does not resize it to 500 pixels or whatever you set...